organiza o código e remove funções não usadas

ramirosevergnini opened the pull request.

ramirosevergnini committed:
exclui função não usada

lfsevergnini reviewed (commented):
Bom trabalho removendo essa função que não é usada!

Como próximos passos, você poderia tentar remover a necessidade de ter as variáveis booleanas "encontrado", pois essa informação é redundante. Isso significa que a classe biblioteca não precisa guardar esses flags, mas sim retornar o objeto procurado nos métodos de busca.

Exemplo: buscarLivroPorTitulo2 é um método mais organizado e flexível do que o buscarLivroPorTitulo. O principal motivo, é que o método 2 retorna uma informação para quem o chama (ex: programa principal) e deixa essa responsabilidade para quem chama a função, não para a classe Biblioteca.

No geral, as classes devem se responsabilzar só por aquilo que elas "sabem". Normalente, uma classe deve saber/ser apenas uma coisa.

Outra boa prática é que as classes de dados/informações não devem ter interação com o usuário, ou seja, não devem fazer input/output. Em vez disso, elas devem receber parâmetros em seus métdos e devolver/retornar as informações que os programas "chamadores" precisam.

No exemplo desse projeto, a classe "Biblioteca" não deve imprimir coisas na tela, mas sim retornar as informações necesseárioas para que o método `main` da classe BibliotecaProjeto consiga fazer o print na tela.

Isso pode não parecer importante, mas lidar com input/output aumenta bastante a complexidade da classe e também diminui a sua flexibilidade. Exemplo: se a classe Biblioteca faz leitura do teclado, como utilizá-la em um programa que tem uma UI (tela) em vez disso? Para evitar esse tipo de problema, e garantir a **reusabilidade** de classes e regras de negócio, é interesasnte manter essas classes com dados ou regras de negócio responsa´veis apenas por isso.

ramirosevergnini committed:
remoção das flags

remoção das flags encontradoLivro e encontradoNome, ajeitei o método buscarUsuario

ramirosevergnini committed:
Retirada de prints

lfsevergnini commented on src/bibliotecaprojeto/Biblioteca.java:
```diff
@@ -21,77 +19,39 @@ public void inserirUsuario(Usuario usuario){
          this.usuarios.add(usuario);
     }
     public void mostrarLivros(){
-        System.out.println("---------------------");
-        System.out.println("LIVROS DA BIBLIOTECA:");
         for (Livro livro : livros){
             System.out.println(livro.getTitulo());
         }
-        System.out.println("---------------------");
-    }   
```
Este método ainda está imprimindo na tela.

Em vez disso, há algumas opções:
1. Retornar a lista de livros e deixar a responsabilidade de imprimir na tela para quem chama (e.g. BibliotecaProjeto.java)
2. Montar uma string com o texto a ser impresso e retornar a string, mas não imprimi-la. Aí, o método chamador fica responsável por imprimir na tela, mas não precisa fazer o loop na lista de livros.

lfsevergnini reviewed (approved):
Bom trabalho!
